java/src/main/native/src/maps_column_view.cu

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/replace.hpp>
+#include <cudf/lists/contains.hpp>
+#include <cudf/lists/detail/contains.hpp>
+#include <cudf/lists/detail/extract.hpp>
+#include <cudf/lists/extract.hpp>
+#include <cudf/lists/lists_column_view.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/types.hpp>
+#include <maps_column_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+namespace cudf::jni {
+
+namespace {
+column_view make_lists(column_view const &lists_child, lists_column_view const &lists_of_structs) {
+  return column_view{data_type{type_id::LIST},
+                     lists_of_structs.size(),
+                     nullptr,
+                     lists_of_structs.null_mask(),
+                     lists_of_structs.null_count(),
+                     lists_of_structs.offset(),
+                     {lists_of_structs.offsets(), lists_child}};
+}
+} // namespace
+
+maps_column_view::maps_column_view(lists_column_view const &lists_of_structs,
+                                   rmm::cuda_stream_view stream)
+    : keys_{make_lists(lists_of_structs.child().child(0), lists_of_structs)},
+      values_{make_lists(lists_of_structs.child().child(1), lists_of_structs)} {
+  auto const structs = lists_of_structs.child();
+  CUDF_EXPECTS(structs.type().id() == type_id::STRUCT,
```

Comment:
Should this also check that the struct column has exactly two children?

Reply:
Shoot, I thought I'd added that check. Will fix.

Reply:
Added additional assertion.